pkg/ingester/ingester.go

```diff
@@ -194,6 +203,19 @@ func New(cfg Config, clientConfig client.Config, limits *validation.Overrides, c
 		return NewV2(cfg, clientConfig, limits, registerer)
 	}
 
+	if cfg.WALConfig.walEnabled {
+		// If WAL is enabled, we don't transfer out the data to any ingester.
+		// Either the next ingester which takes it's place should recover from WAL
+		// or the data has to be flushed during scaledown.
+		cfg.MaxTransferRetries = 0
+
+		recordPool = sync.Pool{
```

Comment:
Instead of a global can we make it part of the WAL itself?

Reply:
I am not sure how I feel about it as it will make some (unnecessary?) additions to the interface. I will add it anyway and we can revert it back if not needed.

Reply:
Ah I see now. Keep it external no issues.